data/mil2525d/utilities/military-features-utilities/Bc2_and_C_CSV_generation_for_domains/Script/GenerateDomainCSV.py

```diff
@@ -0,0 +1,116 @@
+
+    # Table Select
+    if choose_standard == "B2":
+        print "Excluding C tagged items."
+        arcpy.TableSelect_analysis(All_ID_Mapping_Original_Copy, All_ID_Mapping_Sel_10_char, "(CHAR_LENGTH ( \"LegacyKey\" ) = 10) AND (\"Standard\" <> 'C') OR \"Standard\" IS Null")
+    else:
+        print "Excluding B2 tagged items."
+        arcpy.TableSelect_analysis(All_ID_Mapping_Original_Copy, All_ID_Mapping_Sel_10_char, "(CHAR_LENGTH ( \"LegacyKey\" ) = 10) AND (\"Standard\" <> 'B2') OR \"Standard\" IS Null")
+
+    # Load filter_codes_table to in_memory workspace
+    arcpy.CopyRows_management(filter_codes_table, filter_codes_table_Copy, "")
+
+def loop_filter_codes():
+    # Iterate filter_codes_table_Copy to get file names, CharlieName, Filter1, Filter2, FilterGeom values
+    arcpy.AddMessage("Iterating filter code table.")
+    fields = ['CharlieName', 'Filter1', 'Filter2', 'FilterGeom']
+    with arcpy.da.SearchCursor(filter_codes_table_Copy, fields) as cursor:
+        for row in cursor:
+            if row[3] in ["Point", "Line","Area"]:
+                FilterGeom = row[3]
+                the_query = "LegacyKey LIKE " + "\'" + str(row[1])+ "%\'" + " AND " + "GeometryType = " + "\'" + str(FilterGeom) + "\'"
+                
+            elif row[2] is not None:
+                Filter2 = row[2]
+                the_query = "LegacyKey LIKE " + "\'" + str(row[1])+ "%\'" + " OR " + "LegacyKey LIKE " + "\'" + str(row[2])+ "%\'"
+                
+            else:
+                Filter1 = row[1]
+                the_query = "LegacyKey LIKE " + "\'" + str(row[1])+ "%\'"
+
+            CharlieName = row[0]    
+            write_matching_rows_to_csv(CharlieName, the_query)
+                
+
+def write_matching_rows_to_csv(CharlieName, the_query):
+    # Write out the CSV domain table for a given row in the filter_codes_table
+    checkDupList = []
+    arcpy.AddMessage("Writing: " + str(CharlieName))
+    outName = "Coded_Domain_" + str(CharlieName) + ".csv"
+    outFile = os.path.join(out_domain_location, outName)
+    csvOut = open(outFile, "w")
+    writer = csv.writer(csvOut, delimiter=',', lineterminator='\n')
+    writer.writerow(["Name", "Value"])
+    mainFields = ['Name', 'LegacyKey', 'GeometryType', 'Standard']
+    mainCursor = arcpy.da.SearchCursor(All_ID_Mapping_Sel_10_char, mainFields, the_query)
+    for row in mainCursor:
+        if row[1] not in checkDupList:
+            writer.writerow([row[0], row[1]])
+            checkDupList.append(row[1])
+    csvOut.close()
+
+if __name__ == '__main__':
+    filter_codes_table = arcpy.GetParameterAsText(0)
+    All_ID_Mapping_Original_csv = arcpy.GetParameterAsText(1)
+    out_domain_location = arcpy.GetParameterAsText(2)
+    choose_standard = arcpy.GetParameterAsText(3)  #Valid values are B2 or C
+
+    try:
+        load_table_in_memory()
+        
+    except Exception as err:
+            arcpy.AddError(traceback.format_exception_only(type(err), err)[0].rstrip())
```

Comment:
More meaningful error message would be helpful ex. "Could not load tables"

Also looks like a mix of tabs/spaces for whitespace (which I think Pro/Python 3 throws a syntax error for)